### DISCUSSION TITLE
Is possible to work with `Option<_>`?

### DISCUSSION THREAD

**Discussion by xoac** [cornucopia-rs/cornucopia#50] [General]
Hi,
I was curious if is it possible to treat all fields that aren't marked with `NOT NULL` in migration as optional?

**Comment by LouisGariepy** [comment]
Hi @xoac, thank you for opening this issue :smile: 

In general its pretty hard/impossible to determine if the result of some query is nulalble (optional). Some DBMS have some degree of nullability inference, but unfortunately PostgreSQL doesn't have much in this regard.

Bottom line: I'd be willing to review and accept a PR for this feature, but this is very difficult to implement in practice.

For the time being, I'll move this issue to the discussion section so I can keep it open and let other chime in :smile: .

One good thing is that the upcoming release will make the syntax for optional columns more ergonomic  :+1: .
